app/webpacker/components/ContactsPage/UserData.jsx

```diff
@@ -1,24 +1,28 @@
 import React from 'react';
 import { Form } from 'semantic-ui-react';
 import I18n from '../../lib/i18n';
+import { useDispatch, useStore } from '../../lib/providers/StoreProvider';
+import { updateUserData } from './store/actions';
 
-export default function UserData({ formValues, setFormValues }) {
-  const handleFormChange = (_, { name, value }) => setFormValues(
-    { ...formValues, [name]: value },
-  );
+export default function UserData({ userDetails }) {
+  const { userData } = useStore();
+  const dispatch = useDispatch();
+  const handleFormChange = (_, { name, value }) => dispatch(updateUserData(name, value));
+
+  if (userDetails) return null;
```

Comment:
I don't understand this. If there is `userData` present, return early and therefor don't render the form at all?!

Reply:
It won't render the `UserData` part of the form. For example, if logged-in user goes to /contact, they won't have to enter the name & email as we already have that data. At the same time, if not-logged-in user goes to /contact, they have to enter the name & email.

Reply:
Ah, I was confused between `userDetails` and `userData`. Can you find a better name for (current) `userDetails` to make clear that it's essentially just a "pre-loaded version" of `userData`?

Reply:
Done